src/modules/trajectory_generation/interface/src/trajectory_publisher.cpp

```diff
@@ -7,56 +7,77 @@
 #include <std_msgs/msg/string.hpp>
 #include <common_ros2/msg/posture.hpp>
 #include <common_ros2/msg/spline.hpp>
-#include <core_datastructures/Posture.hpp>
+
 #include <spline_generation/cubic_spline_generator.hpp>
+#include <core_datastructures/Posture.hpp>
+#include <iostream>
 
 using namespace std::chrono_literals;
+using trajectory_generation::CubicSplineGenerator;
 
 /* This example creates a subclass of Node and uses std::bind() to register a
 * member function as a callback from the timer. */
 
 class TrajectoryPublisher : public rclcpp::Node
 {
   public:
-    TrajectoryPublisher()
-    : Node("trajectory_publisher"), count_(0)
+    TrajectoryPublisher(core_datastructures::Posture& start, core_datastructures::Posture& goal )
+    : Node("trajectory_publisher"), count_(0), start_(start), goal_(goal)
     {
       publisher_ = this->create_publisher<common_ros2::msg::Spline>("/trajectory", 10);
       timer_ = this->create_wall_timer(
       500ms, std::bind(&TrajectoryPublisher::timer_callback, this));
+
     }
 
+
   private:
     void timer_callback()
     {
-    //   auto message = std_msgs::msg::String();
-    //   message.data = "Hello, world! " + std::to_string(count_++);
-    //   RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
-    //   publisher_->publish(message);
-        
-        auto message = common_ros2::msg::Spline();
-        // message.splin_points
-        for(int i=0; i<10; i++){
-            common_ros2::msg::Posture temp;
-            temp.x =i;
-            temp.y = i;
-            message.spline_points.push_back(temp);
-        }
-        // common_ros2::msg::Spline s;
-        // s.spline_points = test;
-        RCLCPP_INFO(this->get_logger(), "Published");
-        publisher_->publish(message);
+             
+      auto message = common_ros2::msg::Spline();
+     
+      // core_datastructures::Posture start{0,0,0,0}, goal{3,8,M_PI/2,0};
+
+      
+      trajectory_generation::CubicSplineGenerator spl(start_,goal_);
+      std::vector<core_datastructures::Posture> spline = spl.get_spline();
+      for(unsigned int i=0; i<spline.size(); i++){
+        auto temp = common_ros2::msg::Posture();
+        temp.x = spline[i].x;
+        temp.y = spline[i].y;
+        message.spline_points.push_back(temp);
+      }
```

Comment:
Create a new library in common->interface. We can maybe name it as convertors.
It can contain free functions to_domain() and to_ros2() that will perform necessary conversions. 
In future we may have to do more such conversions. We can perform function overloading for such conversions.